packages/lib/src/extensions/BetterClient.ts

```diff
@@ -0,0 +1,181 @@
+	public readonly autoCompleteHandler: AutoCompleteHandler
+	public readonly modalSubmitHandler: ModalSubmitHandler
+	public modals: Collection<string, ModalSubmit>
+	public events: Map<string, EventHandler>
+	public hasteStore: Collection<string, string[]>
+	public sudo: Collection<Snowflake, Snowflake> = new Collection()
+	public localeCache: Collection<Snowflake, Locale> = new Collection()
+	public userChannelCache: Collection<`${Snowflake}-${Snowflake}`, Snowflake>
+	public config: LibConfig
+
+	/**
+	 * __dirname is not in our version of ECMA, so we make do with a shitty fix.
+	 */
+	public readonly __dirname: string
+
+	/**
+	 * Create our client.
+	 * @param options - The options for our client.
+	 */
+	constructor(config: LibConfig) {
+		super(config.clientOptions)
+		this.config = config
+
+		this.__dirname = `${resolve()}/dist`
+
+		this.usersUsingBot = new Set()
+
+		this.applicationCommandHandler = new ApplicationCommandHandler(this)
+		this.applicationCommands = new Collection()
+		this.autoCompleteHandler = new AutoCompleteHandler(this)
+
+		this.textCommandHandler = new TextCommandHandler(this)
+		this.textCommands = new Collection()
+
+		this.buttonHandler = new ButtonHandler(this)
+		this.buttons = new Collection()
+
+		this.dropdownHandler = new DropdownHandler(this)
+		this.dropdowns = new Collection()
+
+		this.modalSubmitHandler = new ModalSubmitHandler(this)
+		this.modals = new Collection()
+
+		this.events = new Map()
+
+		this.hasteStore = new Collection()
+		this.localeCache = new Collection()
+		this.userChannelCache = new Collection()
+
+		this.dropdownHandler.loadFiles()
+		this.buttonHandler.loadFiles()
+		this.applicationCommandHandler.loadFiles()
+		this.textCommandHandler.loadFiles()
+		this.modalSubmitHandler.loadModals()
+		this.loadEvents()
+	}
+
+	/**
+	 * Load all the events in the events directory.
+	 */
+	private async loadEvents() {
+		try {
+			const eventsPath = path.join(this.__dirname, "src", "bot", "events")
+			const eventFileNames = getFiles(eventsPath, "js", true)
+
+			await Promise.all(
+				eventFileNames.map(async (eventFileName) => {
+					try {
+						const filePath = path.join(eventsPath, eventFileName)
+						const fileUrl = `file://${filePath.replace(/\\/g, "/")}`
+
+						const eventFile = await import(fileUrl)
+						const eventName = eventFileName.split(".")[0]
+						const event = new eventFile.default(this, eventName)
+						event.listen()
+
+						this.events.set(event.name, event)
+					} catch (error) {
+						this.log(`Failed to load event file: ${eventFileName} - ${error}`, LogLevel.ERROR)
+					}
+				})
+			)
+		} catch (e) {
+			this.log(`Failed to load files for events handler: ${e}`, LogLevel.WARN)
+		}
+	}
+
+	/**
+	 * Reload all the events in the events directory.
+	 */
+	public reloadEvents() {
+		this.events.forEach((event) => event.removeListener())
+		this.loadEvents()
+	}
+
+	/**
+	 * Log to the console. This function can be overridden to provide a custom logger.
```

Comment:
```suggestion
	 * Log to the console. This function can be overwritten to provide a custom logger.
```

Reply:
The way you do it is by using `override`, so that is the correct terminology here